tests/contrib/django/test_middleware.py

```diff
@@ -63,6 +63,21 @@ def test_middleware_trace_function_based_view(self):
         eq_(span.get_tag('http.url'), '/fn-view/')
         eq_(span.resource, 'tests.contrib.django.app.views.function_view')
 
+    def test_middleware_trace_error_500(self):
+        # ensures that all error 500 are well traced
+        url = reverse('error-500')
+        response = self.client.get(url)
+        eq_(response.status_code, 500)
+
+        # check for spans
+        spans = self.tracer.writer.pop()
+        eq_(len(spans), 1)
+        span = spans[0]
+        eq_(span.get_tag('http.status_code'), '500')
+        eq_(span.get_tag('http.url'), '/error-500/')
+        eq_(span.resource, 'tests.contrib.django.app.views.error_500')
+        assert "ZeroDivisionError: integer division or modulo by zero" in span.get_tag('error.stack')
```

Comment:
I spent all day trying to figure it out and reading the code of django and I think it's not possible to handle the exception of a middleware when this one is thrown during `process_request`. 

Indeed when [loading middlewares](https://github.com/django/django/blob/1.10/django/core/handlers/base.py#L78), django [wraps the `get_response` method to handle exceptions](https://github.com/django/django/blob/1.10/django/core/handlers/exception.py#L36-L43). But we are using the `MiddlewareMixin` class that [calls `process_request` before `get_response`](https://github.com/django/django/blob/1.10/django/utils/deprecation.py#L122-L124). So if an error occurs during `process_request`, the programs just crashes.